framework/archiver/utils.py

```diff
@@ -0,0 +1,79 @@
+from framework.auth import Auth
+
+from framework.archiver import (
+    StatResult, AggregateStatResult,
+    ARCHIVER_FAILURE,
+)
+from framework.archiver.settings import (
+    ARCHIVE_PROVIDER,
+)
+
+def archive_provider_for(node, user):
+    return node.get_addon(ARCHIVE_PROVIDER)
+
+def has_archive_provider(node, user):
```

Comment:
This is for future-proofing, i.e. the internal API for checking if a node has a 'archive_provider' changes.

Imagine we use some third-party service for archiving-- we may then need a user's creds. for that service, etc.